lib/apim_user.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Methods used to add users to API manager Products and Groups.
+ */
+// tslint:disable:no-console
+
+import apiManagementClient = require("azure-arm-apimanagement");
+
+import {
+  SubscriptionContract,
+  UserContract,
+  UserCreateParameters
+} from "azure-arm-apimanagement/lib/models";
+
+import { IResourcesConfiguration } from "./config";
+
+import * as crypto from "crypto";
+
+export interface IUserData extends UserCreateParameters {
+  readonly oid: string;
+  readonly productName: string;
+  readonly groups: ReadonlyArray<string>;
+}
+
+/**
+ *  Assigns a deterministic / predicatable id to the user's subscription.
+ *  Useful in case we want to retrieve it later.
+ */
+export const userIdToSubscriptionId = (userId: string, productName: string) =>
+  // do not use dashes here as it does not play well
+  // together with the version field of the Service entity
+  `${userId}${crypto
```

Comment:
pls add a comment explaining why you're doing this, also why md5? why first 5 chars of the hash?